jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/actions/PurgeQueueAction.kt

```diff
@@ -4,62 +4,90 @@
-            } catch (e: PurgeQueueInProgressException) {
-                LOG.warn { "${selected.queueUrl} already has a query purge in progress" }
-                notifyError(
-                    project = project,
-                    content = message("sqs.purge_queue.failed.60_seconds", selected.queueUrl)
-                )
-                SqsTelemetry.purgeQueue(project, Result.Succeeded, selected.queue.telemetryType())
-            } catch (e: Exception) {
-                LOG.error(e) { "Exception thrown while trying to purge query ${selected.queueUrl}" }
-                notifyError(
-                    project = project,
-                    content = message("sqs.purge_queue.failed", selected.queueUrl)
-                )
-                SqsTelemetry.purgeQueue(project, Result.Failed, selected.queue.telemetryType())
+class PurgeQueueAction(
+    private val project: Project,
+    private val client: SqsClient,
+    private val queue: Queue
+) : DumbAwareAction(message("sqs.purge_queue.action")) {
+    private val edtContext = getCoroutineUiContext()
+
+    override fun actionPerformed(e: AnActionEvent) {
+        ProgressManager.getInstance().run(
+            object : Task.Backgroundable(
+                project,
+                message("sqs.purge_queue.action"),
+                false,
+                ALWAYS_BACKGROUND
+            ) {
+                override fun run(indicator: ProgressIndicator) {
+                    val numMessages = client.approximateNumberOfMessages(queue.queueUrl) ?: 0
+                    val response = runBlocking(edtContext) {
+                        MessageDialogBuilder
+                            .yesNo(
+                                message("sqs.purge_queue.confirm.title"),
+                                message("sqs.purge_queue.confirm", queue.queueName, numMessages)
+                            )
+                            .icon(Messages.getWarningIcon())
+                            .show()
+                    }
+                    if (response != Messages.YES) {
+                        SqsTelemetry.purgeQueue(project, Result.Cancelled, queue.telemetryType())
+                        return
+                    }
+                    try {
+                        client.purgeQueue { it.queueUrl(queue.queueUrl) }
+                        LOG.info { "Started purging ${queue.queueUrl}" }
+                        notifyInfo(
+                            project = project,
+                            title = message("aws.notification.title"),
+                            content = message("sqs.purge_queue.succeeded", queue.queueUrl)
+                        )
+                        SqsTelemetry.purgeQueue(project, Result.Succeeded, queue.telemetryType())
+                    } catch (e: PurgeQueueInProgressException) {
+                        LOG.warn { "${queue.queueUrl} already has a query purge in progress" }
```

Comment:
Nice - good to be intelligent on exceptions that are caught so we can give the customer accurate messaging.